### ISSUE TITLE
reddit - discriminative model vs generative

### ISSUE THREAD

**Issue by nishnik** [nishnik/Paper-Leaf#2]
Discriminative training is a very sensible thing to do when using computers that are too slow to learn a really good generative model of the data. As generative models get better, however, the advantage of discriminative training gets smaller and is eventually outweighed by a major disadvantage: the amount of constraint that the data imposes on the parameters of a discriminative model is equal to the number of bits required to specify the correct labels of the training cases, whereas the amount of constraint for a generative model is equal to the number of bits required to specify the input vectors of the training cases. So when the input vectors contain much more structure than the labels, a generative model can learn many more parameters before it overfits. The benefit of learning a generative model is greatly magnified when there is a large supply of unlabeled speech in addition to the training data that has been labeled by a forced HMM alignment. We do not make use of unlabeled data in this paper, but it could only improve our results relative to purely discriminative approaches.

[Link](https://www.reddit.com/r/MachineLearning/comments/abrtgv/181211446_greedy_layerwise_learning_can_scale_to/ed3fe1r)
